be/src/exec/schema_scanner/schema_views_scanner.cpp

```diff
@@ -0,0 +1,231 @@
+    }
+
+    if (NULL != _param->ip && 0 != _param->port) {
+        RETURN_IF_ERROR(SchemaHelper::get_db_names(*(_param->ip),
+                    _param->port, db_params, &_db_result));
+    } else {
+        return Status::InternalError("IP or port doesn't exists");
+    }
+    return Status::OK();
+}
+
+Status SchemaViewsScanner::fill_one_row(Tuple *tuple, MemPool *pool) {
+    // set all bit to not null
+    memset((void *)tuple, 0, _tuple_desc->num_null_bytes());
+    const TTableStatus& tbl_status = _table_result.tables[_table_index];
+    // catalog
+    {
+        tuple->set_null(_tuple_desc->slots()[0]->null_indicator_offset());
+    }
+    // schema
+    {
+        void *slot = tuple->get_slot(_tuple_desc->slots()[1]->tuple_offset());
+        StringValue* str_slot = reinterpret_cast<StringValue*>(slot);
+        std::string db_name = SchemaHelper::extract_db_name(_db_result.dbs[_db_index - 1]);
+        str_slot->ptr = (char *)pool->allocate(db_name.size());
+        str_slot->len = db_name.size();
+        memcpy(str_slot->ptr, db_name.c_str(), str_slot->len);
+    }
+    // name
+    {
+        void *slot = tuple->get_slot(_tuple_desc->slots()[2]->tuple_offset());
+        StringValue* str_slot = reinterpret_cast<StringValue*>(slot);
+        const std::string* src = &tbl_status.name;
+        str_slot->len = src->length();
+        str_slot->ptr = (char *)pool->allocate(str_slot->len);
+        if (NULL == str_slot->ptr) {
+            return Status::InternalError("Allocate memcpy failed.");
+        }
+        memcpy(str_slot->ptr, src->c_str(), str_slot->len);
+    }
+    // definition
+    {
+        void *slot = tuple->get_slot(_tuple_desc->slots()[3]->tuple_offset());
+        StringValue* str_slot = reinterpret_cast<StringValue*>(slot);
+        const std::string* ddl_sql = &tbl_status.ddl_sql;
+        str_slot->len = ddl_sql->length();
+        str_slot->ptr = (char *)pool->allocate(str_slot->len);
+        if (NULL == str_slot->ptr) {
+            return Status::InternalError("Allocate memcpy failed.");
+        }
+        memcpy(str_slot->ptr, ddl_sql->c_str(), str_slot->len);
+    }
+    // check_option
+    {
+        void *slot = tuple->get_slot(_tuple_desc->slots()[4]->tuple_offset());
+        StringValue* str_slot = reinterpret_cast<StringValue*>(slot);
+        // This is from views in mysql
+        const std::string check_option = "NONE";
+        str_slot->len = check_option.length();
+        str_slot->ptr = (char *)pool->allocate(str_slot->len);
+        if (NULL == str_slot->ptr) {
+            return Status::InternalError("Allocate memcpy failed.");
+        }
+        memcpy(str_slot->ptr, check_option.c_str(), str_slot->len);
+    }
+    // is_updatable
+    {
+        void *slot = tuple->get_slot(_tuple_desc->slots()[5]->tuple_offset());
+        StringValue* str_slot = reinterpret_cast<StringValue*>(slot);
+        // This is from views in mysql
+        const std::string is_updatable = "YES";
```

Comment:
Better set it to "No". Because we do not support DELETE or UPDATE on view.

Reply:
DONE